JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt

```diff
@@ -51,45 +61,88 @@ fun JetnewsApp(
-                val allowDrawerToBeShown = windowSize == WindowSize.Compact
-                val sizeAwareDrawerState = rememberSizeAwareDrawerState(allowDrawerToBeShown)
+            val windowSize = rememberWindowSizeState()
+            val allowDrawerToBeShown = windowSize == WindowSize.Compact
+            val sizeAwareDrawerState = rememberSizeAwareDrawerState(allowDrawerToBeShown)
 
-                ModalDrawer(
-                    drawerContent = {
-                        AppDrawer(
-                            currentRoute = currentRoute,
-                            navigateToHome = navigationActions.navigateToHome,
-                            navigateToInterests = navigationActions.navigateToInterests,
-                            closeDrawer = { coroutineScope.launch { sizeAwareDrawerState.close() } }
-                        )
-                    },
-                    drawerState = sizeAwareDrawerState,
-                    // Only enable opening the drawer via gestures if we allow showing it
-                    gesturesEnabled = allowDrawerToBeShown
-                ) {
-                    JetnewsNavGraph(
-                        appContainer = appContainer,
-                        // Either allow showing the drawer, or show the nav rail
-                        showNavRail = !allowDrawerToBeShown,
-                        navController = navController,
-                        openDrawer = { coroutineScope.launch { sizeAwareDrawerState.open() } },
-                        navigationActions = navigationActions,
+            ModalDrawer(
+                drawerContent = {
+                    AppDrawer(
+                        currentRoute = currentRoute,
+                        navigateToHome = navigationActions.navigateToHome,
+                        navigateToInterests = navigationActions.navigateToInterests,
+                        closeDrawer = { coroutineScope.launch { sizeAwareDrawerState.close() } }
                     )
-                }
+                },
+                drawerState = sizeAwareDrawerState,
+                // Only enable opening the drawer via gestures if we allow showing it
+                gesturesEnabled = allowDrawerToBeShown
+            ) {
+                JetnewsNavGraph(
+                    appContainer = appContainer,
+                    // Either allow showing the drawer, or show the nav rail
+                    showNavRail = !allowDrawerToBeShown,
+                    navController = navController,
+                    openDrawer = { coroutineScope.launch { sizeAwareDrawerState.open() } },
+                    navigationActions = navigationActions,
+                )
             }
         }
     }
 }
 
+/**
+ * Returns the [WindowMetrics] corresponding to the current activities
+ * [WindowInfoRepository.currentWindowMetrics] as [State].
+ */
+@Composable
+private fun rememberCurrentWindowMetrics(): State<WindowMetrics> {
```

Comment:
Nit: What about moving these helper functions to a separate file? These might come at some point from a Compose library

Reply:
Moved to `WindowSize.kt` in the latest commit!